### PR TITLE
Fix wallet address dup key issue

### PR DESCRIPTION
**Pull request by mcornaton**

### CODE REVIEW

File: src/main/java/com/iexec/resultproxy/jwt/JwtService.java [hunk 1]
```diff
@@ -21,16 +27,17 @@ public JwtService(JwtRepository jwtRepository) {
...
         String jwtString;
-        Optional<Jwt> oExistingJwt = findByWalletAddress(signedChallenge.getWalletAddress());
-
-        if (oExistingJwt.isPresent()) {
-            jwtString = oExistingJwt.get().getJwtString(); // TODO generate new token
-        } else {
-            jwtString = Jwts.builder().setAudience(signedChallenge.getWalletAddress()).setIssuedAt(new Date())
-                    .setSubject(RandomStringUtils.randomAlphanumeric(64)).compact();
-            save(new Jwt(signedChallenge.getWalletAddress(), jwtString));
+        synchronized (locks.computeIfAbsent(signedChallenge.getWalletAddress(), key -> new Object())) {
```
Comment (jeremyjams) [inline src/main/java/com/iexec/resultproxy/jwt/JwtService.java:30]: Would you be able to add a note about "why" synchronizing? (I agree we need some synchronized access to this created-or-not singleton per client)

Comment (mcornaton) [reply src/main/java/com/iexec/resultproxy/jwt/JwtService.java:34]: Sure, see https://github.com/iExecBlockchainComputing/iexec-result-proxy/pull/34/commits/01adf197f873f32067c11332223a64da732fa5ab.

Comment (jeremyjams) [reply src/main/java/com/iexec/resultproxy/jwt/JwtService.java:34]: Thanks :)

Comment (jeremyjams) [reply src/main/java/com/iexec/resultproxy/jwt/JwtService.java:34]:
```
     * <p>This implementation assumes that the ConcurrentMap cannot contain null
     * values and {@code get()} returning null unambiguously means the key is
[...]
    @Override
    default V computeIfAbsent(K key,
```
I understand we need to stick with new Object() value if we keep computeIfAbsent() method.

Comment (mcornaton) [reply src/main/java/com/iexec/resultproxy/jwt/JwtService.java:34]: In any way, we need an object to synchronize on. It cannot be a `null` value. So, yes, we need to stick with `new Object()` there.

**Review by jeremyjams — APPROVED** [review]

**Review by jbern0rd — APPROVED** [review]

**Pull request merged**